### PR TITLE
Move current node upgrade tasks under openshift_node_upgrade role

### PR DESCRIPTION
**Pull request by ingvagabund**
Moving some tasks from ``playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml`` into separate role. Preparation for running upgrade nodes play as a part of control plane upgrade over masters.

### CODE REVIEW

**Comment by ingvagabund** [comment]
aos-ci-test

File: roles/openshift_node_upgrade/meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  description: OpenShift Node upgrade
+  company: Red Hat, Inc.
+  license: Apache License, Version 2.0
+  min_ansible_version: 2.1
+  platforms:
+  - name: EL
+    versions:
+    - 7
+dependencies:
+  - role: openshift_common
```
Comment (dgoodwin) [inline roles/openshift_node_upgrade/meta/main.yml]: Arguably openshift_version is a dep here as well, which basically just no-ops if it sees that the three core variables are already set.

File: playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml [hunk 2]
```diff
@@ -43,83 +35,18 @@
     command: >
       {{ hostvars[groups.oo_first_master.0].openshift.common.admin_binary }} drain {{ openshift.node.nodename | lower }} --force --delete-local-data
     delegate_to: "{{ groups.oo_first_master.0 }}"
```
Comment (dgoodwin) [inline playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml]: How does everyone feel about keeping drain / mark schedulable out of the role? This implementation seems best to me as it's probably better, maybe even necessary, to control the ordering when we get to including it in control plane upgrade.

File: roles/openshift_node_upgrade/README.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+Example Playbook
+----------------
+
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
+
+```
+---
+- name: Upgrade nodes
+  hosts: oo_nodes_to_upgrade
```
Comment (dgoodwin) [inline roles/openshift_node_upgrade/README.md]: At some point how this group is calculated should be documented here in the readme, specifically how you can run node upgrade on only nodes with a specific label, and with a value for serial.

File: roles/openshift_node_upgrade/tasks/docker/restart.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  failed_when: false
+  when: openshift.common.is_containerized | bool
+
+- name: Wait for master API to come back online
+  wait_for:
+    host: "{{ openshift.common.hostname }}"
+    state: started
+    delay: 10
+    port: "{{ openshift.master.api_port }}"
+  when: inventory_hostname in groups.oo_masters_to_config
```
Comment (dgoodwin) [inline roles/openshift_node_upgrade/tasks/docker/restart.yml]: This must be the duplication you referred to. (with playbooks/common/openshift-cluster/upgrades/docker/restart.yml) Would this be best solved with an openshift_docker_restart role? (not now but in future)

File: roles/openshift_node_upgrade/tasks/docker/upgrade.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+- debug: var=docker_image_count.stdout
+  when: docker_upgrade_nuke_images is defined and docker_upgrade_nuke_images | bool
+
+- service: name=docker state=stopped
+
+- name: Upgrade Docker
+  package: name=docker{{ '-' + docker_version }} state=present
+
+- include: restart.yml
+  when: not skip_docker_restart | default(False) | bool
```
Comment (dgoodwin) [inline roles/openshift_node_upgrade/tasks/docker/upgrade.yml]: Likewise probably a good candidate for a role later on too.

**Comment by sdodson** [comment]
@dgoodwin are you requesting changes or ok to merge as is?

**Comment by ingvagabund** [comment]
aos-ci-test

File: roles/openshift_node_upgrade/README.md (hunk 3 above)
Comment (ingvagabund) [inline roles/openshift_node_upgrade/README.md]: This is worth doing for each and every role we currently have.

File: roles/openshift_node_upgrade/tasks/docker/restart.yml (hunk 4 above)
Comment (ingvagabund) [inline roles/openshift_node_upgrade/tasks/docker/restart.yml]: * files/nuke_images.sh
* handlers/main.yml
* tasks/containerized_node_upgrade.yml
* tasks/docker/*
* tasks/rpm_upgrade.yml
* tasks/systemd_units.yml
* templates/*

Most of the files is picked from ``playbooks/common/openshift-cluster/upgrades`` directory. Most likely, most of the files can be removed from that. Need to check each place where they are actually used before that. The remaining files are picked from `openshift_node` role. In the former case, the duplicas can be eventually removed. Just the `openshift_node` files will stay.

Comment (ingvagabund) [inline roles/openshift_node_upgrade/tasks/docker/restart.yml]: `openshift_docker_restart role` makes sense to me. Yet, that is something we need to decide globally as part of a bigger re-factoring.

**Comment by mtnbikenc** [comment]
Upgrade from release-1.4 tested.  :+1:

**Pull request merged**